### PR TITLE
Fix/close connection on internal auth error

### PR DESCRIPTION
**Pull request by saadataz**
If we get an internal server error while authenticating, the chat socket should retry connection.
image

### CODE REVIEW

File: src/ws/Socket.ts [hunk 1]
```diff
@@ -684,4 +686,14 @@ export class Socket extends EventEmitter {
             this.status = Socket.CLOSED;
         }
     }
+
+    private callAuth(args: authArgs, options?: ICallOptions) {
+        return this.call(authMethod, args, options).catch((err: IErrorPacket) => {
```
Comment (dadeside) [inline src/ws/Socket.ts]: nit: can this be done with a try/catch and await?

File: src/ws/Socket.ts [hunk 2]
```diff
@@ -14,16 +14,18 @@ import {
...
     UACCESS,
     UnknownCodeError,
     UNOTFOUND,
 } from '../errors';
-import { Reply } from './Reply';
+import { IErrorPacket, Reply } from './Reply';
 
 // The method of the authentication packet to store.
 const authMethod = 'auth';
+type authArgs = [number, number, string, string | undefined];
```
Comment (SimonSchick) [inline src/ws/Socket.ts]: This should be capitalized.

File: src/ws/Socket.ts (hunk 1 above)
Comment (SimonSchick) [inline src/ws/Socket.ts]: Nit: Return type?

File: src/ws/Socket.ts (hunk 2 above)
Comment (connor4312) [reply src/ws/Socket.ts]: And exported!

File: src/ws/Socket.ts (hunk 1 above)
Comment (connor4312) [reply src/ws/Socket.ts]: async/await expands horribly when targeting es5, unfortunately

File: src/ws/Reply.ts [hunk 3]
```diff
@@ -1,22 +1,30 @@
...
-    error: string;
-    data: any;
+  error: string | IErrorPacket;
+  data: any;
+}
+
+export interface IErrorPacket {
+  code: ErrorCode;
+  message: string;
+  stacktrace: string;
```
Comment (connor4312) [inline src/ws/Reply.ts]: We should probably never be returning stacktraces from inside the chat server?

Comment (SimonSchick) [reply src/ws/Reply.ts]: Yeeaaa that shouldn't be the case *al all*.

File: readme.md [hunk 4]
```diff
@@ -16,6 +16,28 @@ We have tutorials on our [Developer Site](https://dev.mixer.com) to help you get
 - [Chat Tutorial](https://dev.mixer.com/tutorials/chatbot.html)
 - [REST API Tutorial](https://dev.mixer.com/tutorials/rest.html)
 
+### Visual Studio Code Reference Launch Config
```
Comment (connor4312) [inline readme.md]: Can you put this under a `Contributing` section or similar? To make it clear to consumers that this is for if they want to iterate on the client itself, and isn't related to any particular debugging setup they need if they're consuming this.

File: src/ws/Reply.ts (hunk 3 above)
Comment (saadataz) [reply src/ws/Reply.ts]: That what we get for some reason
cc: @dadeside ^

**Pull request merged**